core/src/banking_stage.rs

```diff
@@ -985,23 +1049,48 @@ impl BankingStage {
         msgs: &Packets,
         transaction_indexes: &[usize],
         secp256k1_program_enabled: bool,
-    ) -> (Vec<HashedTransaction<'static>>, Vec<usize>) {
-        transaction_indexes
+        cost_model: &Arc<CostModel>,
+        cost_tracker: &Arc<Mutex<CostTracker>>,
+    ) -> (Vec<HashedTransaction<'static>>, Vec<usize>, Vec<usize>) {
+        // make a copy of cost_tracker to statically check limit for this batch of packets
+        let guard = cost_tracker.lock().unwrap();
+        let mut cost_tracker = guard.clone();
+        drop(guard);
```

Comment:
Is there any difference between doing this and just doing:
`let mut cost_tracker = cost_tracker.lock().unwrap();`?

Reply:
it could be `let mut cost_tracker = cost_tracker.lock().unwrap().clone();`

Reply:
add comment to document the intent